src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/MacProxy.cs

```diff
@@ -0,0 +1,125 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+using Microsoft.Win32.SafeHandles;
+using static Interop.CoreFoundation;
+using static Interop.RunLoop;
+
+using CFRunLoopRef = System.IntPtr;
+
+namespace System.Net.Http
+{
+    internal sealed class MacProxy : IWebProxy
+    {
+        public ICredentials Credentials
+        {
+            get => null;
+            set => throw new NotSupportedException();
+        }
+
+        static Uri GetProxyUri(string scheme, CFProxy proxy)
+        {
+            var uriBuilder = new UriBuilder(
+                scheme,
+                proxy.HostName,
+                proxy.PortNumber);
+
+            // FIXME: Handle user name and password
```

Comment:
I'm not sure how is this supposed to be handled correctly. `SocketsHttpHandler` first gets `IWebProxy.Credentials` and then calls `IWebProxy.GetProxy`. I don't have access to the credentials before `GetProxy` methods gets called. I could create `CredentialCache` object (or some custom implementation), return it with `IWebProxy.Credentials` and then add the credentials to the existing `ICredentials` here when `GetProxyUri` is called. The tricky part is to ensure that there are no thread safety problems and that the cache doesn't get too big due to someone constantly changing system settings.

Reply:
How/where/when do you get the credentials?  Is it that the mac interface gives you credentials as part of giving you the proxy uri?

Reply:
>  Is it that the mac interface gives you credentials as part of giving you the proxy uri?

Yes. I have it available exactly at the point of this source code comment.

Reply:
We don't usually have comments in the source code with "FIXME" words.  We either put in a TODO comment with an issue number.

In this case, the interface for implementing a System.Net.IWebProxy interface is to get/set n the credentials via the .Credentials property.  If the Mac internal logic puts the username/password into the URI itself, it is better to strip that out when passing back the Uri.  And to return the credentials as part of the .Credentials property.  This is the contract for implementing the IWebProxy interface.

Reply:
I'll change it to TODO with issue number.

As I already tried to explain above the way that SocketHttpHandler reads the Credentials property makes it nearly impossible to use it. I already strip the username/password during the conversion back to Uri.